### PR TITLE
TUP-703 TACC Home Banner Links Open in New Window

### PR DESCRIPTION
**Pull request by R-Tomas-Gonzalez**
## Overview

Prevent banner links from opening in new windows.

## Related

- [TUP-703](https://tacc-main.atlassian.net/browse/TUP-703)

## Changes

Changes what is considered an external link. Looks for the host name with or without www., then opens in new window if it isn't matching to the document's host.

## Testing

A) If you'd like to troubleshoot in TUP-UI Repo:
1. Comment out code on Core CMS
2. Follow https://github.com/TACC/Core-CMS/wiki/Locally-Develop-CMS-and-TUP-CMS
3. Make snippet with code in TUP -> look at attachment for example
4. Add snippet to Django and import it in the footer

B) If you'd like to just test these changes in TUP-UI using Core_CMS
1. Follow https://github.com/TACC/Core-CMS/wiki/Locally-Develop-CMS-and-TUP-CMS

## UI

…

## Notes 

The `shouldForceSetTarget` function below:

```
      const shouldForceSetTarget = pathsToForceSetTarget.some(path => {
        let shouldForce;
        if (path instanceof RegExp) {
          shouldForce = path.test(link.pathname);
        }
        if (typeof path === 'string') {
          shouldForce = _doPathsMatch(path, link.pathname);
        }
        if (SHOULD_DEBUG && shouldForce) {
          console.debug(`Path "${link.pathname}" matches "${path}"`);
        }
        return shouldForce;
      });
```

does not allow external links to open in a new window. 

For instance. If you're in localhost, and have a `tacc.utexas.edu` link - it should open in a new window because it doesn't match the host. 

 _This might be something you test in tup-ui repo using the A) testing steps above_

### CODE REVIEW

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js [hunk 1]
```diff
@@ -55,14 +55,11 @@ export default function findLinksAndSetTargets(options) {
...
-      const isInternal = (link.host === document.location.host);
-      const shouldSetTarget = shouldForceSetTarget || (
-          ! isInternal && isExternal && ! isMailto
-      );
 
-      if ( shouldSetTarget ) {  
+      const isInternalLink = link.host === baseDocumentHost || link.host === baseDocumentHostWithWWW
+
+      //shouldForceSetTarget could be used in this if statement if passing in options.
+      if (!isInternalLink || isMailto ) {
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:62]: Yup. (Thanks for the comment/reminder.) We avoid `shouldForceSetTarget` for now, cuz it causes the bug.

I'll check that no client needs that feature, and if one does, how to not require this script to handle that.

Comment (wesleyboar) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:62]: **Update:** No client needs `shouldForceSetTarget`. **Celebrate with Fire:** You may burn all code and variables relevant to it e.g. `pathsToExernalSite`, `pathsToForceSetTarget`. **Bonus:** You may clean up cruft code i.e. perform [CMD-87](https://tacc-main.atlassian.net/browse/CMD-87) at your leisure, now or later.

**Review by wesleyboar — COMMENTED** [review]
Looks good. Please delete `shouldForceSetTarget` and related code.

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js [hunk 2]
```diff
@@ -33,13 +33,13 @@ export default function findLinksAndSetTargets(options) {
   }
 
   const links = scopeElement.getElementsByTagName('a');
+  const baseDocumentHost = document.location.host;
+  const baseDocumentHostWithWWW = `www.${baseDocumentHost}`;
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:37]: **Request:** Use a name that suggests relationship to the base document host and/or that this is a boolean e.g. `isSameHostAsBaseDoc…`, `baseDocSameHost…`, `hasBaseDocHost…`.

**Review by wesleyboar — CHANGES_REQUESTED** [review]

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js (hunk 2 above)
Comment (wesleyboar) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:37]: These are not booleans yet. Ignore my dumb. 🫤

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js (hunk 1 above)
Comment (R-Tomas-Gonzalez) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js]: Fixed. [72e3d15](https://github.com/TACC/Core-CMS/pull/805/commits/72e3d15847353f91b0e63751a36cc78bff7a8389)

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js [hunk 3]
```diff
@@ -1,37 +1,4 @@
-/**
- * Whether to log debug info to console
- * @const {string}
- */
-const SHOULD_DEBUG = window.DEBUG;
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:5]: **Woah. Control the burn.** The `SHOULD_DEBUG` is still used. Also, the [jsDoc block](https://jsdoc.app/about-block-inline-tags) (a [docblock](https://en.wikipedia.org/wiki/Docblock)) is still accurate.

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js [hunk 4]
```diff
@@ -1,37 +1,4 @@
...
- */
-
-/**
- * Set external links (automatically discovered) to open in new tab
- * @param {object} [options] - Optional parameters
- * @param {array.<string>} [options.pathsToExernalSite=[]] - (DEPRECATED) A list of relative URL paths that should be treated like external URLs
- * @param {array.<string|RegExp>} [options.pathsToForceSetTarget=[]] - A list of relative URL paths (or patterns) that should trigger setting a target
- * @param {HTMLElement|Document} [options.scopeElement=document] - The element within which to search for links
- * @param {setTargetCallback} [options.setTargetCallback] - A callback for after a target is set
- */
```
Comment (wesleyboar) [inline taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:20]: **Please retain the [docblock](https://en.wikipedia.org/wiki/Docblock) sans `@param` lines.** The function description is still accurate.

<sup>Docblocks are written in [jsDoc format](https://jsdoc.app/about-block-inline-tags). They provide manual definitions (in lieu of TypeScript), and can be used to auto-generate documentation e.g. [vite.js on jsdocs.io](https://www.jsdocs.io/package/vite#RollupCommonJSOptions.transformMixedEsModules).</sup>

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js (hunk 3 above)
Comment (R-Tomas-Gonzalez) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:5]: Sorry. https://apnews.com/article/texas-panhandle-fire-evacuations-cbbb6a279bef1bd020722ed48927114a

Comment (R-Tomas-Gonzalez) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js]: Restored via [cbad538](https://github.com/TACC/Core-CMS/pull/805/commits/cbad538bc59e333040f84fd3e665cb388c71b5af)

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js (hunk 4 above)
Comment (R-Tomas-Gonzalez) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js]: Restored via [cbad538](https://github.com/TACC/Core-CMS/pull/805/commits/cbad538bc59e333040f84fd3e665cb388c71b5af) without params.

File: taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js [hunk 5]
```diff
@@ -12,57 +12,22 @@ const SHOULD_DEBUG = window.DEBUG;
```
Comment (R-Tomas-Gonzalez) [inline taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:12]: @wesleyboar Should I keep lines 7-11 as well?

Comment (wesleyboar) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:12]: Yes please.

Comment (R-Tomas-Gonzalez) [reply taccsite_cms/static/site_cms/js/modules/setTargetForExternalLinks.js:12]: Sounds good. Resolving.

**Review by wesleyboar — APPROVED** [review]

**Pull request merged**